src/libraries/System.ComponentModel.Primitives/ref/System.ComponentModel.Primitives.cs

```diff
@@ -89,12 +89,14 @@ public partial class DescriptionAttribute : System.Attribute
     [System.AttributeUsageAttribute(System.AttributeTargets.Class | System.AttributeTargets.Interface, AllowMultiple=true, Inherited=true)]
     public sealed partial class DesignerAttribute : System.Attribute
     {
-        public DesignerAttribute(string designerTypeName) { }
-        public DesignerAttribute(string designerTypeName, string designerBaseTypeName) { }
-        public DesignerAttribute(string designerTypeName, System.Type designerBaseType) { }
-        public DesignerAttribute(System.Type designerType) { }
-        public DesignerAttribute(System.Type designerType, System.Type designerBaseType) { }
+        public DesignerAttribute([System.Diagnostics.CodeAnalysis.DynamicallyAccessedMembers(System.Diagnostics.CodeAnalysis.DynamicallyAccessedMemberTypes.PublicParameterlessConstructor)] string designerTypeName) { }
```

Comment:
Nice, I didn't know you could add DynamicallyAccessedMembers to a string, thought it only worked for Type.

Reply:
Correct. This way hard-coded strings are recognized by the trimmer, and you can preserve both the Type pointed to by the string, and members on it.